hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,34 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    TableIdentifier baseTableIdentifier = identifier;
+    if (!isValidIdentifier(identifier)) {
+      if (isValidMetadataIdentifier(identifier)) {
```

Comment:
nit: might just be a personal preference but I find the code more readable and compact when we check for exit criteria first, like this:
```
if (!isValidMetadataIdentifier()) {
  return false;
}
baseTableIdentifier = ...;
```